db/migrate/20230828162433_add_access_token_attributes_to_client_configs.rb

```diff
@@ -0,0 +1,5 @@
+class AddAccessTokenAttributesToClientConfigs < ActiveRecord::Migration[6.1]
+  def change
+    add_column :client_configs, :access_token_attributes, :string, array: true
```

Comment:
Would this benefit from having a default value? I know that adding a column with a default value was expensive in the past since it would update all rows on a table to add the default, but they fixed that in PG 11. I'm not sure if there are any other downsides of adding a default in this instance. 

```ruby
add_column :client_configs, :access_token_attributes, :string, array: true, default: %w[first_name last_name email]
```

If we needed to change the default in the future, we could do a quick migration.
```ruby
# Irreversible form
change_column_default :client_configs, :access_token_attributes, %w[first_name last_name email foobar]

# Reversible form
change_column_default :client_configs, :access_token_attributes, from: %w[first_name last_name email], to: %w[first_name last_name email foobar]
```

Reply:
The default should be `nil`. `access_tokens` shouldn't have user attributes unless explicitly requested

Reply:
Gotcha, so this change only applies to the single use case discussed during stand-up--makes sense!

Reply:
I wonder if defaulting to an empty array makes sense. It might preclude some nil checks. Here's a contrived example to show what I mean.

```ruby
# No need to check against nil value with empty array default
def access_token_attributes_as_strings
  client_config.access_token_attributes.map(&:to_s)
end

# Will throw a `NoMethodError` without guard clause checking for nil
def access_token_attributes_as_strings
  return [] unless client_config.access_token_attributes.present?

  client_config.access_token_attributes.map(&:to_s)
end

Reply:
Updated to make an empty array the default value.